integration/spark/app/src/test/java/io/openlineage/spark/agent/ArgumentParserTest.java

```diff
@@ -52,13 +53,17 @@ void testTransportTypes() {
     ArgumentParser argumentParserKinesis =
         ArgumentParser.parse(
             new SparkConf().set(ArgumentParser.SPARK_CONF_TRANSPORT_TYPE, "kinesis"));
+    ArgumentParser argumentParserCustom =
+        ArgumentParser.parse(new SparkConf().set(ArgumentParser.SPARK_CONF_TRANSPORT_TYPE, "test"));
 
     assert (argumentParserConsole.getOpenLineageYaml().getTransportConfig()
         instanceof ConsoleConfig);
     assert (argumentParserHttp.getOpenLineageYaml().getTransportConfig() instanceof HttpConfig);
     assert (argumentParserKafka.getOpenLineageYaml().getTransportConfig() instanceof KafkaConfig);
     assert (argumentParserKinesis.getOpenLineageYaml().getTransportConfig()
         instanceof KinesisConfig);
+    assert (argumentParserCustom.getOpenLineageYaml().getTransportConfig()
```

Comment:
What extra scenario is verified when testing custom transport in Spark? Aren't the tests within client/java sufficient?

Reply:
The test in spark (ArgumentParserTest) tests that the custom transportConfig was resolved correctly by the new TransportConfigTypeIdResolver (which replaces the JsonSubTypes that were previously defined in TransportConfig).
The test in client/java (TransportFactoryTest) tests that the custom transport was created successfully based on the custom config